code/modules/research/designs/weapon_designs.dm

```diff
@@ -172,159 +122,165 @@
-	build_path = /obj/item/ammo_box/magazine/mm712x82/bouncy
+	materials = list(/datum/material/iron = 200)
+	build_path = /obj/item/ammo_box/magazine/ds12g/rb
 	category = list(DCAT_AMMO)
 	mapload_design_flags = DESIGN_FAB_SECURITY | DESIGN_FAB_OMNI
 
-//Rifle
-
-/datum/design/m10mm_rifle
-	name = "10-round magazine (10mm Auto)"
-	desc = "A well-worn magazine fitted for the surplus rifle."
-	id = "m10mm_rifle"
+/datum/design/ds12g/stunslug //Idk what's their difference so i added both
+	name = "Magazine SCL-shotgun stunslug"
+	desc = "A shotgun magazine"
+	id = "ds12g_stunslug"
 	build_type = FABRICATOR
-	materials = list(/datum/material/iron = 350)
-	build_path = /obj/item/ammo_box/magazine/m10mm/rifle
+	materials = list(/datum/material/iron = 250)
+	build_path = /obj/item/ammo_box/magazine/ds12g/stunslug
 	category = list(DCAT_AMMO)
 	mapload_design_flags = DESIGN_FAB_SECURITY | DESIGN_FAB_OMNI
 
-/datum/design/m556
-	name = "30-round magazine (5.56x45mm)"
-	desc = "A well-worn magazine fitted for the surplus rifle."
-	id = "m556"
+/datum/design/ds12g/slug
+	name = "Magazine SCL-shotgun slug"
+	desc = "A shotgun magazine"
+	id = "ds12g_slug"
 	build_type = FABRICATOR
-	materials = list(/datum/material/iron = 350)
-	build_path = /obj/item/ammo_box/magazine/m556
+	materials = list(/datum/material/iron = 300)
+	build_path = /obj/item/ammo_box/magazine/ds12g/slug
 	category = list(DCAT_AMMO)
 	mapload_design_flags = DESIGN_FAB_SECURITY | DESIGN_FAB_OMNI
 
-//SMG
+/datum/design/ds12g/pulseslug
+	name = "Magazine SCL-shotgun pulseslug"
+	desc = "A shotgun magazine"
+	id = "ds12g_pulseslug"
+	build_type = FABRICATOR
+	materials = list(/datum/material/iron = 300)
+	build_path = /obj/item/ammo_box/magazine/ds12g/pulseslug
+	category = list(DCAT_AMMO)
+	mapload_design_flags = DESIGN_FAB_SECURITY | DESIGN_FAB_OMNI
 
-/datum/design/wt550m9
-	name = "20-round magazine (4.6x30mm)"
-	desc = "A WT magazine."
-	id = "wt550m9"
+/datum/design/ds12g
+	name = "Magazine SCL-shotgun buckshot"
+	desc = "A shotgun magazine"
+	id = "ds12g"
 	build_type = FABRICATOR
 	materials = list(/datum/material/iron = 350)
-	build_path = /obj/item/ammo_box/magazine/wt550m9
+	build_path = /obj/item/ammo_box/magazine/ds12g
 	category = list(DCAT_AMMO)
 	mapload_design_flags = DESIGN_FAB_SECURITY | DESIGN_FAB_OMNI
 
-/datum/design/plastikov9mm
-	name = "50-round magazine (9x19mm)"
-	desc = "A plastikov magazine."
-	id = "plastikov9mm"
+
+
+/datum/design/pulse_heavy
+	name = "Heavy Pulse Rifle power cell"
+	desc = "A heavy power pack designed for use with the Heavy Pulse Rifle."
+	id = "pulse_heavy"
 	build_type = FABRICATOR
-	materials = list(/datum/material/iron = 450)
-	build_path = /obj/item/ammo_box/magazine/plastikov9mm
+	materials = list(/datum/material/iron = 9000, /datum/material/silver = 900)
+	build_path = /obj/item/stock_parts/cell/pulse_heavy
 	category = list(DCAT_AMMO)
 	mapload_design_flags = DESIGN_FAB_SECURITY | DESIGN_FAB_OMNI
```

Comment:
The heavy pulse rifle power cell cannot be removed or installed on the gun, it has a recharging mechanic. This should never be made or found under any circumstances.